lib/src/main/kotlin/com/swmansion/starknet/data/serializers/GatewayTransactionTransformingSerializer.kt

```diff
@@ -1,12 +1,14 @@
 package com.swmansion.starknet.data.serializers
 
 import com.swmansion.starknet.data.types.transactions.*
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.json.JsonElement
 import kotlinx.serialization.json.JsonTransformingSerializer
 import kotlinx.serialization.json.jsonObject
 
-internal object GatewayTransactionTransformingSerializer : JsonTransformingSerializer<Transaction>(Transaction.serializer()) {
+internal object GatewayTransactionTransformingSerializer :
+    JsonTransformingSerializer<Transaction>(Transaction.serializer()) {
     override fun transformDeserialize(element: JsonElement): JsonElement {
-        return element.jsonObject["transaction"]!!
+        return element.jsonObject["transaction"] ?: throw SerializationException("Transaction missing in response.")
```

Comment:
Do we still need this? 🤔

Reply:
We do. The way it was previously with `!!` will produce a not very descriptive NullPointerException on any malformed response. Throwing explicitly will make fixing errors a bit simpler in the future.